pkg/proxy/webhook_event_processor.go

```diff
@@ -0,0 +1,222 @@
+	"io"
+	"net/http"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+
+	"github.com/stripe/stripe-cli/pkg/websocket"
+)
+
+// WebhookEventProcessorConfig defines the external inputs that infuence the
+// behavior of a WebhookEventProcessor.
+type WebhookEventProcessorConfig struct {
+	// The logger used to log messages to stdin/err
+	Log *log.Logger
+
+	// List of events to listen and proxy
+	Events []string
+
+	// OutCh is the channel to send logs and statuses to for processing in other packages
+	OutCh chan websocket.IElement
+
+	// Indicates whether to filter events formatted with the default or latest API version
+	UseLatestAPIVersion bool
+
+	// Indicates whether to skip certificate verification when forwarding webhooks to HTTPS endpoints
+	SkipVerify bool
+
+	// Override default timeout
+	Timeout int64
+}
+
+// WebhookEventProcessor encapsulates logic around processing and forwarding
+// webhook events.
+type WebhookEventProcessor struct {
+	cfg *WebhookEventProcessorConfig
+
+	// Events is the supported event types for the command
+	events          map[string]bool
+	endpointClients []*EndpointClient
+	sendMessage     func(*websocket.OutgoingMessage)
+}
+
+// NewWebhookEventProcessor constructs a WebhookEventProcessor from the provided
+// websocket delivery function, route table, and config.
+func NewWebhookEventProcessor(sendMessage func(*websocket.OutgoingMessage), routes []EndpointRoute, cfg *WebhookEventProcessorConfig) *WebhookEventProcessor {
+	p := &WebhookEventProcessor{
+		cfg:         cfg,
+		events:      convertToMap(cfg.Events),
+		sendMessage: sendMessage,
+	}
+
+	for _, route := range routes {
+		// append to endpointClients
+		p.endpointClients = append(p.endpointClients, NewEndpointClient(
+			route.URL,
+			route.ForwardHeaders,
+			route.Connect,
+			route.EventTypes,
+			&EndpointConfig{
+				HTTPClient: &http.Client{
+					CheckRedirect: func(req *http.Request, via []*http.Request) error {
+						return http.ErrUseLastResponse
+					},
+					Timeout: time.Duration(cfg.Timeout) * time.Second,
+					Transport: &http.Transport{
+						TLSClientConfig: &tls.Config{InsecureSkipVerify: cfg.SkipVerify},
+					},
+				},
+				Log:             cfg.Log,
+				ResponseHandler: EndpointResponseHandlerFunc(p.processEndpointResponse),
+				OutCh:           cfg.OutCh,
+			},
+		))
+	}
+
+	return p
+}
+
+// ProcessEvent processes webhook events, notifying listeners via the configured
+// OutCh, sending acknowledgements with the configured websocket sender, and
+// forwarding events to configured endpoints.
+//
+// ProcessEvent implements the websocket.EndpointResponseHandler interface.
+func (p *WebhookEventProcessor) ProcessEvent(msg websocket.IncomingMessage) {
+	if msg.WebhookEvent == nil {
+		p.cfg.Log.Debug("WebSocket specified for Webhooks received non-webhook event")
+		return
+	}
+
+	webhookEvent := msg.WebhookEvent
+
+	p.cfg.Log.WithFields(log.Fields{
+		"prefix":                   "proxy.Proxy.processWebhookEvent",
```

Comment:
Nit: for easier debugging, the prefixes in this file could be updated with the new struct and method names